File: packages/elixir-sdk/lib/treedx_sdk/adapters/exec.ex
defmodule TreeDxSdk.Exec do
  @moduledoc false
  alias TreeDxSdk.Adapters.Common

  def run(client, workspace_id, body) do
    Common.json_request(
      client,
      :post,
      "/api/v1/workspaces/" <> Common.segment(workspace_id) <> "/exec",
      body,
      %{}
    )
  end
end

File: packages/rust-sdk/src/adapters/exec.rs
use std::sync::Arc;

use serde_json::Value;

use crate::adapters::common::{json_request, segment};
use crate::error::TreeDxResult;
use crate::transport::{Transport, TreeDxHttpMethod};

#[derive(Clone)]
pub struct ExecAdapter {
    transport: Arc<dyn Transport>,
}

impl ExecAdapter {
    pub fn new(transport: Arc<dyn Transport>) -> Self {
        Self { transport }
    }

    pub async fn run(&self, workspace_id: &str, body: Value) -> TreeDxResult<Value> {
        json_request(
            &self.transport,
            TreeDxHttpMethod::Post,
            format!("/api/v1/workspaces/{}/exec", segment(workspace_id)),
            Some(body),
            None,
        )
        .await
    }
}
